test/unit/Container.spec.ts

```diff
@@ -98,4 +113,14 @@ describe("Container", function() {
         Container.get(ExtraService).byeMessage.should.be.equal("buy world");
     });
 
+    it("should support class factories", function() {
```

Comment:
Actually, I'm starting to think that better design would be to allow end-user to have multiple separate containers, each with it's own list of registered services and created instances. This will make the container more flexible and it will solve the problem mentioned above for us, i.e. we will be able to just create another container inside of each individual test.

The API could look like this:

```ts
const container1 = Container.create();
container1.registerService(/* Foo */);
container1.registerService(/* Bar */);

const container2 = Container.create();
container2.registerService(/* Baz */);

const bar = container1.get(Bar);
bar.doBar();

const baz = container2.get(Baz);
baz.doBaz();

const foo = container2.get(Foo); // Error: Foo service is not registered

```

I think Symfony is a great example of a quality code, we could try to make a similar API of the Symfony container: http://symfony.com/doc/current/components/dependency_injection.html. As you can see, the container is a normal class, not a static class there: https://github.com/symfony/dependency-injection/blob/master/Container.php.

What do you think about such design? Of course we are talking about breaking changes here, but according to versioning this module is not yet stable ( < 1.0.0).